### ISSUE TITLE
Not in fullscreen

### ISSUE THREAD

**Issue by matttbe** [pdfpc/pdfpc#64]
Hello,

First, thank you for developing and maintaining pdfpc!

I'm using pdfpc 4.0.0 on Ubuntu Wily (15.10). I used packages from official repositories.
With this version, pdfpc is not launched in fullscreen mode: I see a window with the size of my screen's resolution and with a bar on the top. I only tested when using one screen.
Note that launching pdfpc with `-w` option give me two smaller windows.

What can I do to help you fixing this bug? :-)

Regards,

Matt

**Comment by matttbe** [comment]
When adding a second screen, the fullscreen mode is correctly used!

Note that I don't have this bug with one single screen when using pdfpc version 3.1.1.

**Comment by AndreasBilke** [comment]
Maybe it is related to some unity bug?

https://bugs.launchpad.net/unity-mir/+bug/1324824

**Comment by arbox** [comment]
I observe the same behaviour on Ubuntu 15.04 and the pdfpc from the repository.

**Comment by matttbe** [comment]
@AndreasBilke : I don't think so, I'm not using Unity nor Mir.

I also try pdfpc when adding another second screen and this time, I only have this bug with the window on my main screen.

**Comment by rschroll** [comment]
If this is a problem with 4.0 and not 3.1, I wonder if this is related 
to the GTK3 transition (36e0a96).

What window manager / desktop environment are you using?

**Comment by matttbe** [comment]
Compiz is my window manager and the desktop environment is Unity/Gnome (I'm using a Cairo-Dock session)

**Comment by AndreasBilke** [comment]
@matttbe could you try to git-bisect the problem until you find the commit which introduced the bug?

**Comment by matttbe** [comment]
I'll try to do that ASAP (not before the next 2 weeks)!

**Comment by rschroll** [comment]
I bet you a nickel it's 36e0a96.

**Comment by matttbe** [comment]
Hello,

I was able to find the commit which introduces this bug:

> a5105be534897a79bdd674a4b066d4ba204db970 is the first bad commit

**Comment by AndreasBilke** [comment]
Again the CSS commit? So this is maybe related to #68? Actually without
having this issue it is really hard to debug.

**Comment by andybarry** [comment]
@matttbe can you test this with the latest master and see if it is still a problem?  What resolution is your single-screen running at?

**Comment by AndreasBilke** [comment]
@arbox Do you still use ubuntu 15.04? Maybe the problem vanished with some recent ubuntu version :)

**Comment by jakobwenzel** [comment]
This problem also exists on Fedora 23 with Gnome 3.

**Comment by madoar** [comment]
I have the same problem with version v4.0.2 on Linux Mint 17.3 Rosa with Cinnamon

**Comment by perror** [comment]
I got the source from the Github project and compiled it on my Debian and I observed exactly the behavior described here.

Anyway, I solved the problem by doing this simple change in the code (might not be the solution, but it may lead to resolution):

```
diff --git a/src/classes/window/fullscreen.vala b/src/classes/window/fullscreen.vala
index e6f9bcb..2bcc28c 100644
--- a/src/classes/window/fullscreen.vala
+++ b/src/classes/window/fullscreen.vala
@@ -73,7 +73,7 @@ namespace pdfpc.Window {
             // We always render ouput to fit to an exact size.
             // This also forces some tiling window managers like i3 to
             // put the windows on the right screens.
-            this.resizable = false;
+            this.resizable = true;
 
             if (!Options.windowed) {
                 // Move to the correct monitor

```

**Comment by AndreasBilke** [comment]
But pdfpc v4.0.2 (not git master) is not affected? This line of code was
changed in pr #147. Maybe we need to revert this.

**Comment by perror** [comment]
Yes, this is correct. I did not have the bug in v4.0.2, but in git master. Setting `resizable` to true solved the problem.

**Comment by madoar** [comment]
Changing the variable `resizable` from `false` to `true`, fixed the problem for me too.

**Comment by simonnagl** [comment]
I also testet it on cinnamon now with head e4f49ce2385a202e7929c59aef3b59fa97c027ec and agree that my bugfix for i3 introduced a bug for cinnamon.

I wrote a quite minimal example in vala to check how both bugs can be fixed.
You can download it [here](https://github.com/pdfpc/pdfpc/files/222577/fullscreen.txt).

To compile it you need to rename it to fullscreen.vala and then run
``
valac --pkg gtk+-3.0 fullscreen.vala
``

The problem is that i3 wants resizable=false before moving a window to another screen.
This is the right behaviour for this window manager. But I do not know why this must be done before calling show_all().

In cinnamon before calling show_all() the property resizable must be true. Otherwise it will not fullscreen when calling fullscreen(). This behaviour seems strange to me.

I do not know how to fix both bugs. Maby we should contact the developers of i3 and cinnamon. For i3 I had opend [this](http://thread.gmane.org/gmane.comp.window-managers.i3.general/2059/focus=2071) thread on their mailing list.

Do you have some other ideas how to fix that?
@perror Which window manager do you use?

**Comment by arbox** [comment]
@AndreasBilke For me it works fine on v4.0.2. Thank you!
